File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    lst = list(s)
    count = 0
    answer = 0
    for i in lst:
        if count == 0 or check == i:
            check = i
            count += 1
        else :
            count -= 1
            if count == 0:
                answer += 1
    if count > 0:
        answer += 1
    return answer

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
public class Solution {
    public int solution(String s) {
        char[] lst = s.toCharArray(); 
        int count = 0; 
        int answer = 0; 
        char check = '\0'; 

        for (char i : lst) {
            if (count == 0 || check == i) {
                check = i; 
                count++; 
            } else {
                count--; 
                if (count == 0) {
                    answer++; 
                }
            }
        }

        if (count > 0) {
            answer++; 
        }
        
        return answer;
    }
}
